genie-web/src/main/java/com/netflix/genie/web/services/loadbalancers/script/ScriptLoadBalancer.java

```diff
@@ -174,139 +236,139 @@ public Cluster selectCluster(
+                }
 
-            // Download and cache the file (if it's not already there)
-            this.fileTransferService.getFile(scriptFileSource, scriptDestinationPath.toUri().toString());
+                tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_OK));
 
-            final ScriptEngine engine = this.scriptEngineManager.getEngineByExtension(scriptExtension);
-            // We want a compilable engine so we can cache the script
-            if (!(engine instanceof Compilable)) {
-                throw new IllegalArgumentException(
-                    "Script engine must be of type " + Compilable.class.getName()
+                this.isConfigured.set(true);
+            } catch (final GenieException | IOException | ScriptException | RuntimeException | URISyntaxException e) {
+                tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_FAILED));
+                tags.add(Tag.of(MetricsConstants.TagKeys.EXCEPTION_CLASS, e.getClass().getName()));
+                log.error(
+                    "Refreshing the load balancing script for ScriptLoadBalancer failed due to {}",
+                    e.getMessage(),
+                    e
                 );
+                this.isConfigured.set(false);
+            } finally {
+                this.isUpdating.set(false);
+                this.registry
+                    .timer(UPDATE_TIMER_NAME, tags)
+                    .record(System.nanoTime() - updateStart, TimeUnit.NANOSECONDS);
+                log.debug("Refresh completed");
             }
-            final Compilable compilable = (Compilable) engine;
-            try (
-                InputStream fis = Files.newInputStream(scriptDestinationPath);
-                InputStreamReader reader = new InputStreamReader(fis, UTF_8)
-            ) {
-                log.debug("Compiling {}", scriptFileSource);
-                this.script.set(compilable.compile(reader));
-            }
-
-            tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_OK));
+        }
 
-            this.isConfigured.set(true);
-        } catch (final GenieException | IOException | ScriptException | RuntimeException | URISyntaxException e) {
-            tags.add(Tag.of(MetricsConstants.TagKeys.STATUS, STATUS_TAG_FAILED));
-            tags.add(Tag.of(MetricsConstants.TagKeys.EXCEPTION_CLASS, e.getClass().getName()));
-            log.error(
-                "Refreshing the load balancing script for ScriptLoadBalancer failed due to {}",
-                e.getMessage(),
-                e
-            );
-            this.isConfigured.set(false);
-        } finally {
-            this.isUpdating.set(false);
-            this.registry
-                .timer(UPDATE_TIMER_NAME, tags)
-                .record(System.nanoTime() - updateStart, TimeUnit.NANOSECONDS);
-            log.debug("Refresh completed");
+        @Nullable
+        CompiledScript get() {
```

Comment:
Is this one of those `@VisibleForTesting` things? Seems unnecessary

Reply:
It's part of the script `Loader` interface so it can be mocked easily.
But yeah, package private is probably not the best choice, i'll fix it.